tests/functional/ovndbcluster_controller_test.go

```diff
@@ -27,11 +28,152 @@ import (
 	condition "github.com/openstack-k8s-operators/lib-common/modules/common/condition"
 	. "github.com/openstack-k8s-operators/lib-common/modules/common/test/helpers"
 	"github.com/openstack-k8s-operators/ovn-operator/api/v1beta1"
+	ovnv1 "github.com/openstack-k8s-operators/ovn-operator/api/v1beta1"
 	corev1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/types"
 )
 
 var _ = Describe("OVNDBCluster controller", func() {
+	When("OVNDBCluster CR is created with replicas, dns info should be present",
+		func() {
+			DescribeTable("should create and delete dnsdata CR",
+				func(DnsName string) {
+					var clusterName types.NamespacedName
+					var OVNSBDBClusterName types.NamespacedName
+					var OVNNBDBClusterName types.NamespacedName
+					var cluster *ovnv1.OVNDBCluster
+					var dbs []types.NamespacedName
+					_ = th.CreateNetworkAttachmentDefinition(types.NamespacedName{Namespace: namespace, Name: "internalapi"})
+					dbs = CreateOVNDBClusters(namespace, map[string][]string{namespace + "/internalapi": {"10.0.0.1"}}, 3)
+					OVNNBDBClusterName = types.NamespacedName{Name: dbs[0].Name, Namespace: dbs[0].Namespace}
+					OVNSBDBClusterName = types.NamespacedName{Name: dbs[1].Name, Namespace: dbs[1].Namespace}
+					cluster = GetOVNDBCluster(OVNNBDBClusterName)
+					clusterName = OVNNBDBClusterName
+					if DnsName == "sb" {
+						cluster = GetOVNDBCluster(OVNSBDBClusterName)
+						clusterName = OVNSBDBClusterName
+					}
+					statefulSetName := types.NamespacedName{
+						Namespace: cluster.Namespace,
+						Name:      "ovsdbserver-" + DnsName,
+					}
+
+					Eventually(func(g Gomega) int {
+						listDns := GetDNSDataList(statefulSetName)
+						return len(listDns.Items)
+					}).Should(BeNumerically("==", 3))
+
+					// Scale down to 1
+					Eventually(func(g Gomega) {
+						c := GetOVNDBCluster(clusterName)
+						*c.Spec.Replicas = 1
+						g.Expect(k8sClient.Update(ctx, c)).Should(Succeed())
+					}).Should(Succeed())
+
+					// Check if dnsdata CR is down to 1
+					Eventually(func() int {
+						targetedDnsOcurrences := 0
+						listDns := GetDNSDataList(statefulSetName)
```

Comment:
don't you have a label to filter out relevant DNSData entries now? I think you should use the label instead of manual Contains check.

Reply:
It would be better indeed. Done